### DISCUSSION TITLE
Transpose

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#47] [Q&A] [answered]
Hi, is possible to transpose all notes not on keyboard, but on squishbox?
My alesis vortex wireless 2 doesn't have this function...
Thanks.

**Comment by albedozero** [comment, marked as answer]
Sure, just add a `router_rule` in your bank file(s) like this:
```yaml
- {type: note, chan: 1=2, par1: 0-127*1+5}
```
This will transpose all incoming notes on channel 1 up 5 steps and send them to whatever sound you've assigned to channel 2. See the [router rules lesson](https://youtu.be/53OFZnKhPa8?t=332) for another example.

The Vortex Wireless 2 does have octave up/down buttons on the neck, but I assume you're asking about transposing by amounts other than a full octave. Router rules allow you to do this, and you can make it specific to certain patches.

**Reply by tizfra** [reply]
I've read something like that in bank 0, is possible to use a fader to increase and decrease transpose?

**Reply by albedozero** [reply]
Yes, but it's a bit more complicated. One way would be to use special CC messages called registered parameter numbers (RPNs) to set the coarse tuning value. If you have a fader that sends CC#15 on channel 1, for example, you could use these router rules to do want you want:
```yaml
- {type: cc, chan: 1=2, par1: 15=101, par2: 0-127=0}
- {type: cc, chan: 1=2, par1: 15=100, par2: 0-127=2}
- {type: cc, chan: 1=2, par1: 15=6, par2: 0-127=52-76}
```
Every time you move the fader, the first two rules select RPN#2, which is coarse tuning. The third rule then sets the number of semitones to adjust the tuning. A value of 64 is zero semitones, so this rule will adjust the tuning +/- 12 steps. The `chan` parameter in the rules lets you control which MIDI channel you are transposing.

This works differently from the `note` rule [above](https://github.com/albedozero/fluidpatcher/discussions/47#discussioncomment-3073027) - it actually retunes the synthesizer on that channel. Because of this, if you transpose by a large amount the sound quality will suffer because the samples are playing in note range for which they weren't intended. If you are a python programmer, then a better way to get this behavior would be to add your own special router rule parameter that would be detected by the [_listener()_](https://github.com/albedozero/fluidpatcher/blob/11bb4d06dbc494b4c245e70889472b8ddb4a009b/squishbox.py#L147) function in _squishbox.py_ and then add note router rules using [_add_router_rule()_](https://github.com/albedozero/fluidpatcher/blob/11bb4d06dbc494b4c245e70889472b8ddb4a009b/patcher/__init__.py#L263).

**Reply by tizfra** [reply]
It works like a charm
Thank you
